app/javascript/bundles/RestaurantsApp/components/AddRestaurant.jsx

```diff
@@ -53,49 +50,65 @@ export default class AddRestaurantForm extends React.Component {
     submit(model) {
         this.setState({isFormSubmitted: true});
         if (this.state.isFormValid) {
-            this.setState({isAddingRestaurant: true});
-            const restaurant = {
-                restaurant: {
-                    name: model.restaurantName,
-                    speed: model.speed.value || null,
-                    cuisine_types: model.cuisineType ? [model.cuisineType.value] : [],
-                    accepts_10bis: model.accepts10bis || false,
-                    kosher: model.isKosher || false,
-                }
-            };
+            this.setState({isSubmitting: true});
+            const {
+                onAdd,
+                onClose
+            } = this.props;
```

Comment:
I usually put this at the top of the class, just because it's easier to find when skimming the code.